src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -2109,8 +2111,12 @@ def x25519_load_private_bytes(self, data):
         # contains an OCTET STRING of length 32! So the last two bytes here
         # are \x04\x20, which is an OCTET STRING of length 32.
         pkcs8_prefix = b'0.\x02\x01\x000\x05\x06\x03+en\x04"\x04 '
-        bio = self._bytes_to_bio(pkcs8_prefix + data)
-        evp_pkey = backend._lib.d2i_PrivateKey_bio(bio.bio, self._ffi.NULL)
+        with self._zeroed_bytearray(48) as ba:
+            ba[0:16] = pkcs8_prefix
+            ba[16:] = data
```

Comment:
Is data's length actually checked before this?

Reply:
It was not being properly checked and #4703 adds the check (along with making x25519 and x448 consistent across their from_{public,private}_bytes methods